File: app/src/main/java/com/zm/audio/java/utils/TimeUtil.java
package com.zm.audio.java.utils;

/**
 * Created by ZhouMeng on 2018/9/4.
 * 数字转时分秒
 */
public class TimeUtil {
    public static String formatLongToTimeStr(int time) {
        int hour = 0;
        int minute = 0;
        int second = time;
        if (second > 60) {
            minute = second / 60;
            second %= 60;
        }

        if (minute > 60) {
            hour = minute / 60;
            minute %= 60;
        }
        return (hour < 10 ? "0" + hour : hour) + ":" + (minute < 10 ? "0" + minute : minute) + ":"
                + (second < 10 ? "0" + second : second);
    }
}

File: app/src/main/java/com/zm/audio/kotlin/utils/TimeUtil.kt
package com.zm.audio.kotlin.utils

/**
 * Created by ZhouMeng on 2018/9/4.
 * 数字转时分秒
 */

object TimeUtil {
    fun formatLongToTimeStr(time: Int): String {
        var hour = 0
        var minute = 0
        var second = time
        if (second > 60) {
            minute = second / 60         //取整
            second %= 60         //取余
        }

        if (minute > 60) {
            hour = minute / 60
            minute %= 60
        }
        return ((if (hour < 10) "0$hour" else hour).toString() + ":" + (if (minute < 10) "0$minute" else minute) + ":"
                + if (second < 10) "0$second" else second)
    }
}
